go/geth_adapter/adapter.go

```diff
@@ -480,7 +481,22 @@ func (a *runContextAdapter) IsSlotInAccessList(addr vm.Address, key vm.Key) (add
 }
 
 func (a *runContextAdapter) HasSelfDestructed(addr vm.Address) bool {
-	return a.evm.StateDB.HasSuicided(gc.Address(addr))
+	return a.evm.StateDB.HasSelfDestructed(gc.Address(addr))
+}
+
+func uint256ToValue(value *uint256.Int) (result vm.Value, err error) {
+	if value == nil {
+		return result, fmt.Errorf("unable to convert nil to Hash")
+	}
+	if value.ByteLen() > 32 {
```

Comment:
This should be impossible considering the input being a 256 bit value. Considering this, and to return 0 for a nil value, would allow to eliminate the error return parameter. This would enable the function to be used in expressions.